Add steps to read email following copy card order

andrewhick opened the pull request:
These tests now check the email content when registration cards are ordered from the back office. It is the first Waste Carriers test to use the new `/email/last-email` path, to view the latest email sent by the service.

The feature also adds tests under the new payments functionality to copy cards.

The check currently reads the whole email JSON without parsing it, but I intend to reuse the parsing from WEX in future email tests.

New tests and rubocop pass.

cintamani commented on features/page_objects/journey/last_email_page.rb:
```diff
@@ -0,0 +1,32 @@
+      next unless page_text.include?(expected_string)
+
+      return page_text
+    end
```
Suggestion:
```
    10.times do
      page_text = email_content.text

      return page_text if page_text.include?(expected_string)

      page.evaluate_script "window.location.reload()"
    end
```
This will make sure that you only reload the page after you checked the content. The current version will reload the first page without checking the text. 
Also, in general, working with `if` rather than `unless` makes the code more readable :) 
Is a small change but I think it will give a little more chances to get the right email :)

cintamani commented on features/step_definitions/journey/order_cards_steps.rb:
```diff
@@ -60,9 +62,27 @@
+
+  # Get the text of the last email containing the chosen text.
+  # Take care that this test does not pick up an email from a previous test run.
+  # This test could fail if a similar copy card order has been placed recently.
```
NIT: If you check for the content of the email text to include `@registration_number` rather than `"We’re printing your waste carriers registration card"` you should avoid it failing because it picked a similar copy card order

andrewhick commented:
Thanks for the comments. I've replaced an unless with an if, moved the page reload, and moved the assertion code inside the method. Hopefully a bit cleaner now!

cintamani commented on features/page_objects/journey/last_email_page.rb:
```diff
@@ -0,0 +1,43 @@
+        end
+      end
+
+      return true if email_contains_all_text == true
```
NIT: 
```suggestion
      return true if email_contains_all_text
```
